src/formats/onenote.ts

```diff
@@ -142,150 +147,174 @@ export class OneNoteImporter extends FormatImporter {
+		if ('sectionGroups' in entity) {
+			for (const sectionGroup of entity.sectionGroups as SectionGroup[]) {
+				let sectionGroupDiv = parentEl.createDiv(
+					{
+						attr: {
+							style: 'padding-inline-start: 1em; padding-top: 8px'
+						}
+					});
+				
+				sectionGroupDiv.createEl('strong', {
 					text: sectionGroup.displayName!,
 				});
 
-				// Set the parent section group for neater folder structuring
-				sectionGroup.sections?.forEach(section => section.parentSectionGroup = sectionGroup);
-				this.createSectionList(sectionGroup.sections!, sectionDiv);
+				this.renderHierarchy(sectionGroup, sectionGroupDiv);
 			}
 		}
-	}
-
-	createSectionList(sections: OnenoteSection[], parentEl: HTMLDivElement) {
-		const list = parentEl.createEl('ul', {
-			attr: {
-				style: 'padding-inline-start: 1em;',
-			},
-		});
-		for (const section of sections) {
-			const listElement = list.createEl('li', {
-				cls: 'task-list-item',
+	  
+		if ('sections' in entity) {
+			const sectionList = parentEl.createEl('ul', {
+				attr: {
+					style: 'padding-inline-start: 1em;',
+				},
 			});
-			let label = listElement.createEl('label');
-			let checkbox = label.createEl('input');
-			checkbox.type = 'checkbox';
-
-			label.appendChild(document.createTextNode(section.displayName!));
-			label.createEl('br');
-
-			// Add/remove a section from this.selectedSections
-			checkbox.addEventListener('change', () => {
-				if (checkbox.checked) this.selectedSections.push(section);
-				else {
-					const index = this.selectedSections.findIndex((sec) => sec.id === section.id);
-					if (index !== -1) {
-						this.selectedSections.splice(index, 1);
+			for (const section of entity.sections as OnenoteSection[]) {
+				const listElement = sectionList.createEl('li', {
+					cls: 'task-list-item',
+				});
+				let label = listElement.createEl('label');
+				let checkbox = label.createEl('input');
+				checkbox.type = 'checkbox';
+
+				label.appendChild(document.createTextNode(section.displayName!));
+				label.createEl('br');
+
+				checkbox.addEventListener('change', () => {
+					if (checkbox.checked) this.selectedIds.push(section.id!);
+					else {
+						const index = this.selectedIds.findIndex((sec) => sec === section.id);
+						if (index !== -1) {
+							this.selectedIds.splice(index, 1);
+						}
 					}
-				}
-			});
+				});
+
+				this.renderHierarchy(section, parentEl);				
+			}
 		}
 	}
 
 	async import(progress: ImportContext): Promise<void> {
-		// Remove possible duplicates, eg. when the user selects "Select all" with existing selections.
-		this.selectedSections = this.selectedSections.filter((item, index, array) => array.indexOf(item) === index);
-
-		let outputFolder = await this.getOutputFolder();
-		let remainingSections = this.selectedSections.length;
+		// Remove possible duplicate IDs, eg. when the user selects "Select all" with existing selections
+		this.selectedIds =  [...new Set(this.selectedIds)];
```

Comment:
Can we prevent duplicates from happening? Otherwise if there's a duplicate the checkbox won't do anything since it will only add/remove one of the entries for that ID.